src/localCurrency/saga.ts

```diff
@@ -55,23 +52,6 @@ export function* fetchLocalCurrencyRateSaga() {
   }
 }
 
-export function* watchFetchCurrentRate() {
-  yield* takeLatest(Actions.FETCH_CURRENT_RATE, safely(fetchLocalCurrencyRateSaga))
-}
-
-export function* watchSelectPreferredCurrency() {
-  yield* put(fetchCurrentRate())
-  while (true) {
-    yield* take([
-      Actions.SELECT_PREFERRED_CURRENCY,
-      AccountActions.SET_PHONE_NUMBER,
-      AppActions.PHONE_NUMBER_VERIFICATION_COMPLETED,
-    ])
-    yield* put(fetchCurrentRate())
-  }
-}
-
 export function* localCurrencySaga() {
-  yield* spawn(watchFetchCurrentRate)
-  yield* spawn(watchSelectPreferredCurrency)
+  yield* takeLatest([Actions.SELECT_PREFERRED_CURRENCY], safely(fetchLocalCurrencyRateSaga))
```

Comment:
supernit: keeping the watch- prefixed saga seems easier to read to me

Reply:
hum, but that means from the root saga we spawn this localCurrencySaga and all it does is spawn another function that executes this line? 🙈 i'm not sure i understand why the extra layer of abstraction increases readability, is it because we use the `watch*` pattern sometimes? we've not been so consistent with that pattern though

Reply:
yeah, i noticed we use `watch*` pattern quite frequently and it seemed to me as a standard "building block". the abstraction could help to skim right away what this saga is doing: oh, it spawns a "watcher", that's it.

Reply:
okay i put the watcher function back. i don't see a lot of value in the particular pattern 🙈 but i do see value in the consistency